src/lib/features/feature-toggle/feature-toggle-service.ts

```diff
@@ -1049,9 +1050,16 @@ class FeatureToggleService {
     async getPlaygroundFeatures(
         query?: IFeatureToggleQuery,
     ): Promise<FeatureConfigurationClient[]> {
-        const result = await this.clientFeatureToggleStore.getPlayground(
+        let result = await this.clientFeatureToggleStore.getPlayground(
             query || {},
         );
+
+        if (this.flagResolver.isEnabled('separateAdminClientApi')) {
+            result = (await this.getFeatureToggles(
+                query,
+            )) as IFeatureToggleClient[];
```

Comment:
will clean this up when we remove the feature flag

Reply:
shouldn't we do one query or the other? Here we're fetching getPlayground first and then override it with another query

Reply:
If we wanted to optimise this maximally, we could do that. But since this is the admin API and this feature flag should be removed asap to avoid holding all this context, I think it's fine.